game/missiongenerator/aircraft/waypoints/recoverytanker.py

```diff
@@ -0,0 +1,85 @@
+from dcs.point import MovingPoint
+from dcs.task import (
+    ActivateBeaconCommand,
+    RecoveryTanker,
+    Tanker,
+    SetUnlimitedFuelCommand,
+)
+
+from game.ato import FlightType
+from game.utils import feet, knots
+from .pydcswaypointbuilder import PydcsWaypointBuilder
+
+
+class RecoveryTankerBuilder(PydcsWaypointBuilder):
+    def add_tasks(self, waypoint: MovingPoint) -> None:
+
+        assert self.flight.flight_type == FlightType.REFUELING
+
+        # Unlimited fuel option : disable at racetrack start. Must be first option to work.
+        if self.flight.squadron.coalition.game.settings.ai_unlimited_fuel:
+            if waypoint.tasks and isinstance(
+                waypoint.tasks[0], SetUnlimitedFuelCommand
+            ):
+                waypoint.tasks[0] = SetUnlimitedFuelCommand(False)
+            else:
+                waypoint.tasks.insert(0, SetUnlimitedFuelCommand(False))
+
+        waypoint.add_task(Tanker())
+
+        group_id = self._get_carrier_group_id()
+        speed = knots(250).meters_per_second
+        altitude = feet(6000).meters
+
+        # Last waypoint has index of 1.
+        # Give the tanker a end condition of the last carrier waypoint.
+        # If the carrier ever gets more than one waypoint this approach needs to change.
+        last_waypoint = 2
+        recovery_tanker = RecoveryTanker(group_id, speed, altitude, last_waypoint)
+
+        waypoint.add_task(recovery_tanker)
+
+        self.configure_tanker_tacan(waypoint)
+
+    def _get_carrier_group_id(self) -> int:
+        name = self.package.target.name
+        carrier_position = self.package.target.position
+        theater_objects = self.unit_map.theater_objects
+        for key, value in theater_objects.items():
+            # Check name and position in case there are multiple of same carrier.
+            if name in key and value.theater_unit.position == carrier_position:
+                return value.dcs_group_id
+        raise RuntimeError(
+            f"Could not find a carrier in the mission matching {name} at "
+            f"({carrier_position.x}, {carrier_position.y})"
+        )
+
+    def configure_tanker_tacan(self, waypoint: MovingPoint) -> None:
```

Comment:
I think it is needed since it looks like refueling tasks get it through: https://github.com/dcs-retribution/dcs-retribution/blob/dev/game/missiongenerator/aircraft/waypoints/racetrack.py

Reply:
Right my bad, it's the waypoint-action to actually activate the TCN, but since the logic looks quite similar, I was thinking we could maybe refactor it?